lib/vector-core/buffers/src/variant/disk_and_memory.rs

```diff
@@ -0,0 +1,95 @@
+use crate::WhenFull;
+#[cfg(test)]
+use quickcheck::{Arbitrary, Gen};
+use std::path::PathBuf;
+
+#[cfg(test)]
+const MAX_STR_SIZE: usize = 128;
+#[cfg(test)]
+const ALPHABET: [&str; 27] = [
+    "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s",
+    "t", "u", "v", "w", "x", "y", "z", "_",
+];
+
+#[derive(Debug, Clone)]
+pub enum Variant {
```

Comment:
The term `Variant` seems pretty generic here. The use of this name appears to be isolated to the `buffers` crate, so it's probably not a big deal though. Since they are isolated, do they need to be `pub`?

Reply:
They do need to be public so that the benchmark code can use them. I'm open to names other than `Variant`.